WordPress/Classes/ViewRelated/Reader/ReaderPostCardCell.swift

```diff
@@ -0,0 +1,580 @@
+        preserveStartingConstraintConstants()
+        if contentProvider != nil {
+            configureCell(contentProvider!)
+        }
+    }
+
+    /**
+        Ignore taps in the card margins
+    */
+    public override func hitTest(point: CGPoint, withEvent event: UIEvent?) -> UIView? {
+        if (!CGRectContainsPoint(cardContentView.frame, point)) {
+            return nil
+        }
+        return super.hitTest(point, withEvent: event)
+    }
+
+    public override func prepareForReuse() {
+        super.prepareForReuse()
+        applyHighlightedEffect(false, animated: false)
+    }
+
+    public override func sizeThatFits(size: CGSize) -> CGSize {
+        let innerWidth = innerWidthForSize(size)
+        let innerSize = CGSize(width: innerWidth, height: CGFloat.max)
+
+        var height = cardContentView.frame.minY
+
+        height += featuredMediaView.frame.minY
+        height += featuredMediaHeightConstraint.constant
+        height += featuredMediaBottomConstraint.constant
+
+        height += titleLabel.sizeThatFits(innerSize).height
+        height += titleLabelBottomConstraint.constant
+
+        height += summaryLabel.sizeThatFits(innerSize).height
+        height += summaryLabelBottomConstraint.constant
+
+        // The attribution view's height constraint is to be less than or equal
+        // to the constant. Skip the math when the constant is zero, but use
+        // the height returned from sizeThatFits otherwise.
+        if attributionHeightConstraint.constant > 0 {
+            height += attributionView.sizeThatFits(innerSize).height
+        }
+        height += attributionBottomConstraint.constant
+
+        // On the iPad, the tag and word count views are horizontal,
+        // aligned with the action buttons. Only add their heights
+        // for the iPhone.
+        if !UIDevice.isPad() {
+            height += tagButtonHeightConstraint.constant
+            height += tagButtonBottomConstraint.constant
+
+            height += wordCountLabel.sizeThatFits(innerSize).height
+            height += wordCountBottomConstraint.constant
+        }
+
+        height += actionButtonViewHeightConstraint.constant
+        height += actionButtonViewBottomConstraint.constant
+
+        height += cardContentBottomConstraint.constant
+
+        return CGSize(width: size.width, height: height)
+    }
+
+    private func innerWidthForSize(size: CGSize) -> CGFloat {
+        var width = CGFloat(0.0)
+        var horizontalMargin = headerView.frame.minX
+
+        if UIDevice.isPad() {
+            width = maxIPadWidthConstraint.constant
+        } else {
+            width = size.width
+            horizontalMargin += cardContentView.frame.minX
+        }
+        width -= (horizontalMargin * 2)
+        return width
+    }
+
+
+    // MARK: - Configuration
+
+    private func preserveStartingConstraintConstants() {
+        featuredMediaHeightConstraintConstant = featuredMediaHeightConstraint.constant
```

Comment:
Is this a port from another class in ObjC ? (I did see this snippet in another PR, earlier this week)